library/Icinga/Application/Hook.php

```diff
@@ -275,21 +277,23 @@ public static function all($name)
     /**
      * Get the first hook
      *
-     * @param   string  $name   One of the predefined hook names
+     * @param  string  $name              One of the predefined hook names
+     * @param  bool    $bypassPermission  Don't check for permissions
      *
-     * @return  null|mixed
+     * @return mixed|null
      */
-    public static function first($name)
+    public static function first($name, $bypassPermission = false)
     {
         $name = self::normalizeHookName($name);
 
         if (self::has($name)) {
             foreach (self::$hooks[$name] as $key => $hook) {
-                if (self::hasPermission($hook)) {
+                if ($bypassPermission || self::hasPermission($hook)) {
```

Comment:
That sounds dangerous. Where is that used?

Reply:
Within the RequirementHook, reason is that `setup` doesn't use the normal role/permission model

Reply:
Ah ok, understood, thanks 👍